Plausible/Chamelean/DeriveConstrainedProducer.lean

```diff
@@ -490,28 +516,34 @@ def getScheduleForInductiveRelationConstructor
       let countChecks (schd : List ScheduleStep) : Nat :=
         schd.foldl (fun acc step => match step with | .Check _ _ => acc + 1 | _ => acc) 0
 
+      let countUnconstrained (schd : List ScheduleStep) : Nat :=
+        schd.foldl (fun acc step => match step with | .Unconstrained .. => acc + 1 | _ => acc) 0
+
       let mut minChecks := countChecks fstSchd
+      let mut minUnconstrained := countUnconstrained fstSchd
       let mut countSeen  := 1
       let mut minLen     := fstSchd.length
       let mut bestSchedule   := fstSchd
 
       let prefixSize := 100000
 
-      trace[plausible.deriving.arbitrary] m!"First Schedule: {scheduleStepsToString bestSchedule} \nChecks & Length: {(minChecks, minLen)} \nSchedules Considered: {repr countSeen}"
+      trace[plausible.deriving.arbitrary] m!"First Schedule: {scheduleStepsToString bestSchedule} \nChecks & Length & Unconstrained: {(minChecks, minLen, minUnconstrained)} \nSchedules Considered: {repr countSeen}"
 
       for schdM in LazyList.take prefixSize rest.get do
         let schd ← schdM
         let checkCount := countChecks schd
+        let unconstrained := countUnconstrained schd
         countSeen := countSeen + 1
         let len := schd.length
-        if checkCount < minChecks || (checkCount == minChecks && len < minLen) then
+        if checkCount < minChecks || (checkCount == minChecks && len < minLen) || (checkCount == minChecks && len == minLen && unconstrained < minUnconstrained) then
```

Comment:
Can we lift this boolean predicate out maybe? Not sure but it's getting complex and I suspect we'll want to tweak it further.

I guess you can have it be a function of the `(checks, len, constrained)` triple (here it's just lex, I think? In which case you can actually use `Prod.lex`, maybe).